kernel/kernel-api/src/main/java/io/delta/kernel/expressions/PredicateEvaluator.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (2023) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.expressions;
+
+import java.util.Optional;
+
+import io.delta.kernel.annotation.Evolving;
+import io.delta.kernel.data.ColumnVector;
+import io.delta.kernel.data.ColumnarBatch;
+
+/**
+ * Special interface for evaluating {@link Predicate} on input batch and return a selection
+ * vector containing one value for each row in input batch indicating whether the row has passed
+ * the predicate or not.
+ *
+ * Optionally it takes an existing selection vector along with the input batch for evaluation.
```

Comment:
Maybe mention: it should only update (become more selective, go from true to false) for rows that are currently selected (true in the optional selection vector).

i.e. if a row is already marked as `false`, yet it passes this next predicate, we shouldn't allow it to go from `false` to `true`.

This seems obvious, but we might as well document it.